github/src/test/java/io/syndesis/github/GitHubServiceITCase.java

```diff
@@ -62,13 +68,44 @@ public void before() throws IOException {
 
         Assertions.assertThat(authToken).isNotNull().isNotBlank();
 
-        // Setup credentials for GitHub client (Token should include "repo" and "delete_repo" scopes)
-        client = new GitHubClient();
-        client.setOAuth2Token(authToken);
+        client = new KeycloakProviderTokenAwareGitHubClient();
 
         // Now that we have a client we create one of our GitHubService
         githubService = new GitHubServiceImpl(new RepositoryService(client), new UserService(client), gitWorkflow);
 
+        webserver = new DefaultMockServer();
+        webserver.start(1234);
+
+        webserver.expect().get().withPath("/auth/realms/syndesis-it/broker/github/token").andReturn(
+            200,
+            "{\"access_token\":\"" + authToken + "\",\"expires_in\":2678400,\"scope\":\"user:info user:check-access role:edit:syndesis-staging:! role:system:build-strategy-source:syndesis-staging\",\"token_type\":\"Bearer\"}"
+        )
+            .withHeader("Content-Type", "application/json")
+            .always();
+
+        KeycloakAuthenticationToken keycloakAuthenticationToken = new KeycloakAuthenticationToken(
+            new SimpleKeycloakAccount(
+                new KeycloakPrincipal<RefreshableKeycloakSecurityContext>("testuser", null),
+                null,
+                new RefreshableKeycloakSecurityContext(
+                    null,
+                    null,
+                    "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWUsImlzcyI6Imh0dHA6Ly9sb2NhbGhvc3Q6MTIzNC9hdXRoL3JlYWxtcy9zeW5kZXNpcy1pdCJ9.N0R7v55eiVXMzzhooYb1BStzyu_iP07wKNIO9z9dRMs",
```

Comment:
It's a dummy JWT we use for the test - you can inspect the contents at https://jwt.io/